runtime/src/main/cpp/ObjCExport.mm

```diff
@@ -943,8 +945,21 @@ static void throwIfCantBeOverridden(Class clazz, const KotlinToObjCMethodAdapter
       insertOrReplace(methodTable, adapter->nameSignature, const_cast<void*>(adapter->kotlinImpl));
       RuntimeAssert(adapter->vtableIndex == -1, "");
 
-      if (adapter->itableIndex != -1 && superITable != nullptr)
+      if (adapter->itableIndex != -1 && superITable != nullptr) {
+        // Because of fake overrides [adapter->interfaceId] might not be equal to [typeInfo->classId_].
+        auto interfaceId = adapter->interfaceId;
+        int interfaceVTableSize = adapter->itableSize;
+        RuntimeAssert(interfaceVTableSize >= 0, "");
+        auto interfaceVTablesIt = interfaceVTables.find(interfaceId);
+        if (interfaceVTablesIt == interfaceVTables.end()) {
+          interfaceVTables.emplace(interfaceId, KStdVector<VTableElement>(interfaceVTableSize));
+        } else {
+          auto const& interfaceVTable = interfaceVTablesIt->second;
+          RuntimeAssert(interfaceVTable.size() == static_cast<size_t>(interfaceVTableSize), "");
+        }
```

Comment:
Looks pretty much like a copy-paste from the code above. It should be possible to extract a (lambda) function for this code.